sdk/go/property/type.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package property
+
+type valueType uint8
+
+const (
```

Comment:
As in the other comment I don't think this is needed. If a value is "computed/unknown" it doesn't have a type and in all other cases you can just ask the rtti system for the type of the value object.